openedx/core/djangoapps/credentials/tests/test_utils.py

```diff
@@ -0,0 +1,114 @@
+    def test_get_credentials_client_initialization_failure(self, mock_init):
+        """Verify behavior when API client fails to initialize."""
+        mock_init.side_effect = Exception
+
+        actual = get_user_credentials(self.user)
+        self.assertEqual(actual, [])
+        self.assertTrue(mock_init.called)
+
+    @httpretty.activate
+    def test_get_user_credentials_retrieval_failure(self):
+        """Verify behavior when data can't be retrieved from Credentials."""
+        self.mock_api(self.url, self.CREDENTIALS_API_RESPONSE, status_code=500)
+
+        actual = get_user_credentials(self.user)
+        self.assertEqual(actual, [])
+
+    def test_get_user_program_credentials_issuance_disable(self):
+        """Verify that user program credentials cannot be retrieved if issuance is disabled."""
+        self.create_credential_config(enable_learner_issuance=False)
+        actual = get_user_program_credentials(self.user)
+        self.assertEqual(actual, [])
+
+    @httpretty.activate
+    def test_get_user_program_credentials_no_credential(self):
+        """Verify behavior if no credential exist."""
+        self.mock_api(self.url, {'results': []})
+        actual = get_user_program_credentials(self.user)
+        self.assertEqual(actual, [])
+
+    @httpretty.activate
+    def test_get_user_program_credentials_revoked(self):
+        """Verify behavior if credential revoked."""
+        credential_data = {"results": [
+            {
+                "id": 1,
+                "username": "test",
+                "credential": {
+                    "credential_id": 1,
+                    "program_id": 1
+                },
+                "status": "revoked",
+                "uuid": "dummy-uuid-1"
+            }
+        ]}
+        self.mock_api(self.url, credential_data)
+        actual = get_user_program_credentials(self.user)
+        self.assertEqual(actual, [])
+
+    @httpretty.activate
+    def test_get_user_program(self):
+        """Verify program credentials data can be retrieved and parsed correctly."""
+        credentials_config = self.create_credential_config()
+        program_config = self.create_program_config()
+        program_url = program_config.internal_api_url.strip('/') + '/programs/'
+        self.mock_api(program_url, self.PROGRAMS_API_RESPONSE)
+        self.mock_api(self.url, self.CREDENTIALS_API_RESPONSE, reset_uri=False)
+        actual = get_user_program_credentials(self.user)
+        expected = self.PROGRAMS_API_RESPONSE['results']
+        expected[0]['credential_url'] = \
+            credentials_config.public_service_url + 'credentials/' + self.PROGRAMS_CREDENTIALS_DATA[0]['uuid']
+        expected[1]['credential_url'] = \
+            credentials_config.public_service_url + 'credentials/' + self.PROGRAMS_CREDENTIALS_DATA[1]['uuid']
+        self.assertEqual(len(actual), 2)
+        self.assertEqual(actual, expected)
+
+        httpretty.reset()
```

Comment:
(here) https://github.com/edx/edx-platform/pull/11064/files#diff-3ad46a3d64782ff5370992231aa31486R80 we are not resetting url as if we do so urls that are register in `mock_programs_api()` are also reset and hence `programs api` will not be mocked so once the test ran we should reset the httpretty urls.